server/liveblog/themes/themes.py

```diff
@@ -304,20 +307,32 @@ def on_updated(self, updates, original):
             self.publish_related_blogs(original)
 
     def on_delete(self, deleted_theme):
+        theme = deleted_theme['name']
         global_default_theme = get_resource_service('global_preferences').get_global_prefs()['theme']
         # raise an exception if the removed theme is the default one
-        if deleted_theme['name'] == global_default_theme:
+        if theme == global_default_theme:
             raise SuperdeskApiError.forbiddenError('This is a default theme and can not be deleted')
         # raise an exception if the removed theme has children
-        if self.get(req=None, lookup={'extends': deleted_theme['name']}).count() > 0:
+        if self.get(req=None, lookup={'extends': theme}).count() > 0:
             raise SuperdeskApiError.forbiddenError('This theme has children. It can\'t be removed')
         # update all the blogs using the removed theme and assign the default theme
         blogs_service = get_resource_service('blogs')
-        blogs = blogs_service.get(req=None, lookup={'blog_preferences.theme': deleted_theme['name']})
+        blogs = blogs_service.get(req=None, lookup={})
         for blog in blogs:
-            # will assign the default theme to this blog
-            blog['blog_preferences']['theme'] = global_default_theme
-            blogs_service.system_update(ObjectId(blog['_id']), {'blog_preferences': blog['blog_preferences']}, blog)
+            blog_prefences = blog['blog_preferences']
+            themes_settings = blog['themes_settings']
+            public_urls = blog['public_urls']
+            if blog_prefences['theme'] == theme:
+                # will assign the default theme to this blo
+                blog_prefences['theme'] = global_default_theme
```

Comment:
as before, no previous assignments are necessary. It would be better to let `blog['blog_preferences']['theme']` instead of saving `blog['blog_preferences'] into a new variable`